statev2/src/replication/network.rs

```diff
@@ -0,0 +1,88 @@
+//! The network abstraction of the replication layer; defines the interface for
+//! sending and receiving raft messages from cluster peers
+
+use raft::prelude::Message as RaftMessage;
+
+use super::error::ReplicationError;
+
+/// The central trait that must be implemented by any networking layer used
+/// by the replication layer
+pub trait Network {
+    /// The error type emitted by the network implementation
+    type Error: Into<ReplicationError>;
+
+    /// Send a raft message to a peer
+    ///
+    /// This method MAY NOT not block
+    fn send(&mut self, message: RaftMessage) -> Result<(), Self::Error>;
+
+    /// Receive a raft message from the network
+    ///
+    /// This method MAY block
+    fn recv(&mut self) -> Result<Option<RaftMessage>, Self::Error>;
```

Comment:
thoughts on making this `async`?

Reply:
The consensus engine isn't async so we get nothing out of laying an asynchronous runtime on top